### PR TITLE
Processes/higgs processes generalized

### PR DESCRIPTION
**Pull request by mafrahm**
This process adds all combinations of single Higgs production and decay channel processes (around 450 processes in total).
To accomplish this, I implemented helper functions (e.g. `add_decay_process`) that allows creating sub-processes with a single line of code. This makes our process highly generalized and allows us to easily accomplish changes in the naming, IDs, and labels of lots of processes by changing a single value in a dictionary.

Open questions:

- The labels of some single-H processes could be a bit shorter (e.g. `H_{ggf}, H \rightarrow bb` for Process h_ggf_hbb. We could skip the `, H` part of the label, but I'm not sure how to generalize this)
- There still are more H production modes (e.g. bbH, ttVH, tHq, tHW) that are not yet fully implemented, but I assigned them the IDs from 110000 to 169999
- some (probably irrelevant) Higgs decay modes are not yet considered (e.g. ee and gluglu, and BSM single H). They could be added to our ID scheme e.g. at 910, 920, ...
- I added the xsecs for the remaining H production modes, but I was unable to find the xsecs for ttzh and ttwh.

### CODE REVIEW

File: cmsdb/constants/__init__.py [hunk 1]
```diff
@@ -55,8 +55,15 @@
...
 br_h_gg_full = Number(0.002270, {"theo": (0.0173j, 0.0172j), "m_q": (0.0093j, 0.0099j), "alpha_s": (0.0061j, 0.0062j)})
+br_h_zg_full = Number(0.001533, {"theo": (0.571j, 0.571j), "m_q": (0.0098j, 0.0101j), "alpha_s": (0.0058j, 0.0065j)})
+br_h_gluglu_full = Number(0.08187, {"theo": (0.0340j, 0.0341j), "m_q": (0.0112j, 0.0113j), "alpha_s": (0.0369j, 0.0361j)})  # noqa
 br_h_bb_full = Number(0.5824, {"theo": (0.0065j, 0.0065j), "m_q": (0.0072j, 0.0074j), "alpha_s": (0.0078j, 0.0080j)})
+br_h_cc_full = Number(0.02891, {"theo": (0.0120j, 0.0120j), "m_q": (0.0526j, 0.0098j), "alpha_s": (0.0125j, 0.0125j)})
 br_h_tt_full = Number(0.06272, {"theo": (0.0117j, 0.0116j), "m_q": (0.0098j, 0.0099j), "alpha_s": (0.0062j, 0.0062j)})
+br_h_mm_full = Number(0.0002176, {"theo": (0.0123j, 0.0123j), "m_q": (0.0097j, 0.0099j), "alpha_s": (0.0059j, 0.0064j)})
+
+# TODO: H -> ee BR is not available in the YellowReport, so I'm using the muon BR scaled to electron mass instead
+br_h_ee_full = br_h_mm_full / 200 ** 2
```
Comment (riga) [inline cmsdb/constants/__init__.py:66]: Nice. That's also what https://arxiv.org/pdf/1410.6679 used.

File: cmsdb/processes/higgs.py [hunk 2]
```diff
@@ -61,14 +526,40 @@
...
+h_ggf_hgg = add_decay_process(h_ggf, h_decay_map.hgg)
+h_ggf_hmm = add_decay_process(h_ggf, h_decay_map.hmm)
 
-h_ggf_tautau = h_ggf.add_process(
-    name="h_ggf_tautau",
-    id=11100,
-    xsecs=multiply_xsecs(h_ggf, const.br_h.tt),
-)
+# Higgs sub-decay channels
+# TODO: mapping of parent processes does not yet work here
```
Comment (riga) [inline cmsdb/processes/higgs.py:544]: What is different here w.r.t., for instance, `h_ggf_htt` above?

**Review by riga — DISMISSED** [review]
Really nice, and a lot of work!

I'll approve but I would wait for @nprouvost to agree as well.

Just a side note: some of the blocks in which you define higgs subprocesses look fairly similar and just a detail like the name of the decay map changes. Could there be a way to also loop over that? It might not be particularly clean, but to add the subprocess with their correct names to the module, one could do `locals()[the_name] = add_decay_process(...)`.

**Review by nprouvost — CHANGES_REQUESTED** [review]
just a few changes but else LGTM, thanks for the hard work!

File: cmsdb/processes/higgs.py [hunk 3]
```diff
@@ -261,8 +1498,10 @@
             "pdf": 0.03j,
             "scale": (0.06j, 0.23j),
         }),
-        14: Number(34.43,
-    {"pdf": 0.03j, "scale": (0.06j, 0.23j)})},  # fb
+        14: Number(34.43, {
+            "pdf": 0.03j,
+            "scale": (0.06j, 0.23j),
+        })},  # fb
```
[the remark this answers is not in the document]
Comment (riga) [reply cmsdb/processes/higgs.py:1504]: True, `pb` should be the default.

To keep things readable, I'd suggest `0.001 * Number(34.43, ...`.

**Comment by mafrahm** [comment]
> Really nice, and a lot of work!
> 
> I'll approve but I would wait for @nprouvost to agree as well.
> 
> Just a side note: some of the blocks in which you define higgs subprocesses look fairly similar and just a detail like the name of the decay map changes. Could there be a way to also loop over that? It might not be particularly clean, but to add the subprocess with their correct names to the module, one could do `locals()[the_name] = add_decay_process(...)`.

Yes, it's always the same block adding all the Higgs decay channels and their subsequent decay channels. I propose we do this in a separate PR after this.

File: cmsdb/processes/higgs.py (hunk 2 above)
Comment (mafrahm) [reply cmsdb/processes/higgs.py:544]: Oh I think the comment is just outdated. This has been solved by adding the `production_mode_parent` aux to each process.

**Review by nprouvost — APPROVED** [review]
LGTM

**Review by riga — APPROVED** [review]

**Pull request merged**